### DISCUSSION TITLE
Compiling abstractions

### DISCUSSION THREAD

**Discussion by luismrguimaraes** [charlesneimog/pd4web#19] [Q&A 🇺🇸] [answered]
I created an abstraction and put it under `Libs/`. I then used it in two subpatches, but `pd4web` is not able to compile it. I added `[declare -path Libs/]` in my patch, and it works in Pure Data. The error message is `ValueError: Library or Object can't be processed, please report: [dryWet] inside /Users/<username>/Documents/Code/<project-name>/src/Audio/main.pd`, but even adding the abstraction in the patch doesn't fix it.

**Comment by charlesneimog** [comment]
For now, abstractions files must be inside `Libs` folder and must be created as `Libs/myabs`. 

Things like `declare -path Libs` and then trying to create `myabs` will not work!

Thanks, that was not clear!

**Reply by luismrguimaraes** [reply]
Alright, that fixes it! I did however get execution errors with [else/knob]. Compilation worked fine, but the execution crashed.

**Reply by luismrguimaraes** [reply]
I'm also unable to load an audio file, even though I respected the folder naming, and even used `Audios/audiofile.wav`.

Edit: I get the error `error: [soundfiler] read: ./Audios/sharp.wav: No such file or directory` even though it is there and loads in Pure Data. (This is using the `[else/sample~]` object)

**Reply by charlesneimog** [reply]
All the external GUI features are still in progress, so hold off on using `knob` for now. I’m working on it, and with the 'dev' version, you can compile, but it won't actually run yet.

Are you loading the files with a loadbang? If not try it, I'm currently solving some internal Emscripten issues with using files from the Audio Worklet (60d56f152df8bb16553b55cb16386cbb08af4161), so try deleting the build folder and rebuilding.

**Reply by luismrguimaraes** [reply]
I don't think `[else/sample~]` reads bangs, but even with a loadbang and a message like 
<img width="377" alt="image" src="https://github.com/user-attachments/assets/2386e447-9c5a-41e0-b7ee-05c0a14f08a1">

I get the same error (`[wav file not found](error: [soundfiler] read: ./Audios/sharp.wav: No such file or directory)`).

**Reply by luismrguimaraes** [reply]
(I first tried just adding a loadbang, but it didn't fix the error)

**Reply by charlesneimog** [reply, marked as answer]
For now, try using internal objects like `soundfiler`.

The Audio Worklet provides excellent low latency and improved performance, but there are some complex nuances that are challenging to grasp. To address this, I'll likely need to explore how `sample~` handles file searches and how the abstractions will work inside the audio worklet.

**Reply by luismrguimaraes** [reply]
Alright, thanks!
